maths/prime_number.rb

```diff
@@ -0,0 +1,39 @@
+# A ruby program to check a given number is prime or not
+# Mathematical explanation: A number which has only 2 factors i.e., 1 (one) and itself
+
+# Prime number check function
+def prime_number(number)
+  non_prime_flag = false
+  if number > 0
+    if number == 1
+      non_prime_flag = true
+    elsif number == 2
+      non_prime_flag = false
+    elsif number % 2 == 0
+      non_prime_flag = true
+    else
+      non_prime_flag = (2..Math.sqrt(number)).any? { |i| number % i == 0 }
+    end
+  else
+    non_prime_flag = true
+  end
```

Comment:
I think it is mostly good!

How about flattening the `if` clause?

```suggestion
  if number <= 1
    non_prime_flag = true
  elsif number == 2
    non_prime_flag = false
  elsif number % 2 == 0
    non_prime_flag = true
  else
    non_prime_flag = (2..Math.sqrt(number)).any? { |i| number % i == 0 }
  end
```

Reply:
I have implemented these review comments; can you please do a review again?